pycbc/workflow/pegasus_workflow.py

```diff
@@ -272,17 +277,15 @@ def root_path(v):
                 path += [v.in_workflow]
                 v = v.in_workflow
             return path
-
         workflow_root = root_path(self)
         input_root = root_path(inp)
-
         for step in workflow_root:
             if step in input_root:
                 common = step
                 break
-
-        dep = dax.Dependency(child=workflow_root[workflow_root.index(common)-1],
-                             parent=input_root[input_root.index(common)-1])
+        dep = dax.Dependency(
+            parent=input_root[input_root.index(common)-1].as_job,
+            child=self.as_job)
```

Comment:
this should be 

```
child=workflow_root[workflow_root.index(common) - 1]
```
i.e. as it was before with one other change mentioned above. Your change is fine for what works now in terms of file handling, but is actually not generically the right dependency to add.